capa/features/extractors/ts/engine.py

```diff
@@ -0,0 +1,114 @@
+from typing import List, Tuple, Iterator
+
+from tree_sitter import Node, Tree, Parser
+
+import capa.features.extractors.ts.sig
+import capa.features.extractors.ts.build
+import capa.features.extractors.ts.query
+from capa.features.address import FileOffsetRangeAddress
+from capa.features.extractors.ts.query import QueryBinding
+
+
+class TreeSitterExtractorEngine:
+    query: QueryBinding
+    path: str
+    buf: bytes
+    tree: Tree
+    import_signatures: set
+
+    def __init__(self, language: str, path: str):
+        capa.features.extractors.ts.build.ts_build()
+        self.language = language
+        self.query = capa.features.extractors.ts.query.QueryBinding(language)
+        self.import_signatures = capa.features.extractors.ts.sig.load_import_signatures(language)
+        self.path = path
+        with open(self.path, "rb") as f:
+            self.buf = f.read()
+        self.parse()
+
+    def get_language(self):
+        return self.language
+
+    def get_ts_language(self):
+        return self.query.language
+
+    def parse(self):
+        self.parser = Parser()
+        self.parser.set_language(self.get_ts_language())
+        self.tree = self.parser.parse(self.buf)
```

Comment:
`self.tree` is not referenced/declared in `__init__`.

i'd also generally like to minimize the number of mutating routines, so would you return the tree and let the caller (`__init__`) assign it?